### PR TITLE
Add postgresql support

### PR DESCRIPTION
**Pull request by lieut-data**
This updates our load testing tool with support for Postgres. Note that we're not using any kind of ORM abstraction here, so the changes are a bit more manual than perhaps normal.

Note the individual commits, since there's a lot of vendor/ changes here mostly to fix something wrong with the dep-generated files.

Along the way, I've fixed a handful of issues that tripped up running the load tests and debugging problems, accompanied by a [server change](https://github.com/mattermost/mattermost-server/pull/8491) with other minor changes.

### CODE REVIEW

File: loadtest/bulkload.go [hunk 1]
```diff
@@ -412,15 +413,27 @@ func LoadPosts(cfg *LoadTestConfig, dbendpoint string) {
...
-	for i := 0; i < 100; i++ {
-		statementStr += "(?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?),"
+	if driverName == "mysql" {
+		for i := 0; i < 100; i++ {
+			statementStr += "(?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?),"
+		}
+	} else if driverName == "postgres" {
+		for i := 0; i < 100; i++ {
+			statementStr += fmt.Sprintf(
+				"($%d, $%d, $%d, $%d, $%d, $%d, $%d, $%d, $%d, $%d, $%d, $%d, $%d, $%d, $%d, $%d, $%d, $%d),",
```
Comment (lieut-data) [inline loadtest/bulkload.go:423]: `github.com/lib/pq` doesn't support `?`. I didn't look much harder for other Postgres drivers, but opinions welcome :)

File: loadtest/bulkload.go [hunk 2]
```diff
@@ -439,32 +452,31 @@ func LoadPosts(cfg *LoadTestConfig, dbendpoint string) {
...
+		for page := 0; numReceived == 200; page++ {
+			cmdlog.Infof("Page %d", page)
 			if newusers, resp2 := adminClient.GetUsersInTeam(team.Id, page, 200, ""); resp2.Error != nil {
 				cmdlog.Errorf("Could not get users in team %v. Error: %v", team.Id, resp2.Error.Error())
 				return
-			} else {
-				numRecieved = len(newusers)
-				total += numRecieved
-				cmdlog.Info("User %v", newusers[0].Username)
+			} else if numReceived = len(newusers); numReceived > 0 {
```
Comment (lieut-data) [inline loadtest/bulkload.go:477]: Previously, if we failed to get users, `newusers[0].Username` would panic.

File: loadtest/local_command_runner.go [hunk 3]
```diff
@@ -33,6 +33,7 @@ func (c *MattermostLocalConnection) RunCommand(command string) (bool, string) {
 	cmd := exec.Command(split[0], split[1:]...)
 	var b bytes.Buffer
 	cmd.Stdout = &b
+	cmd.Stderr = cmd.Stdout
```
Comment (lieut-data) [inline loadtest/local_command_runner.go:36]: Stderr often contained the real failure.

**Comment by lieut-data** [comment]
I'd also appreciate if you could check out this branch and just smoke test things locally to ensure my `dep` generated changes aren't causing any issues. I'm somewhat surprised by the line count change...

**Comment by crspeller** [comment]
Wah. That's a lot of deletions. Will check it out.

**Review by crspeller — APPROVED** [review]
Looks great! Thanks for the cleanup. The dep stuff works so I guess all those deletions are fine.

**Pull request merged**